kivy/core/text/markup.py

```diff
@@ -752,7 +752,14 @@ def p_restricted(line, uw, c):
         s = self.get_extents(last_text)
         if len(last_text):
             line1.append(LayoutWord(last_word.options, s[0], s[1], last_text))
-        elps.options = last_word.options
+
+        elps_opts = old_opts
```

Comment:
This stuff needs to happen before `elps_s = textwidth('...')` above because that's when we compute the ellipse size. If the ellipse changes size according to these new opts, we'll have to know it there to make sure it fits.